modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/viewmodel/PlayerViewModel.kt

```diff
@@ -618,14 +615,24 @@ class PlayerViewModel @Inject constructor(
         playbackManager.skipToPreviousChapter()
     }
 
-    fun onChapterClick(chapter: Chapter) {
-        launch {
-            val listData = listDataLive.value
-            if (listData?.isSameChapter(chapter) == true) {
-                _showPlayerFlow.emit(Unit)
-            } else {
-                playbackManager.skipToChapter(chapter)
+    // taping on the chapter title on the now playing screen should scroll to that chapter when the fragment is available
+    var scrollToChapterListener: ((Chapter) -> Unit)? = null
+        set(value) {
+            val chapter = cacheScrollToChapter
+            if (chapter != null) {
+                value?.invoke(chapter)
+                cacheScrollToChapter = null
             }
+            field = value
+        }
+
+    private var cacheScrollToChapter: Chapter? = null
+
+    fun scrollToChapter(chapter: Chapter) {
+        if (scrollToChapterListener == null) {
+            cacheScrollToChapter = chapter
+        } else {
+            scrollToChapterListener?.invoke(chapter)
```

Comment:
I found it quite tricky to open the chapter fragment and scroll to the correct chapter. Let me know if there is a cleaner way to do this.

Reply:
This is tricky! The first few things I tried were complete failures, but I eventually came up [with this](https://github.com/Automattic/pocket-casts-android/commit/4b43f28171aff8a153982050217b985367ff5b22) as an alternative way to accomplish this. I think it's maybe a bit cleaner, but I'm not sure if there might be some edge cases it doesn't cover.

Reply:
Wow, thanks for improving the code. That worked well when I tested this.